packages/devtools_app/lib/src/eval_on_dart_library.dart

```diff
@@ -30,36 +30,43 @@ class EvalOnDartLibrary {
   EvalOnDartLibrary(
     Iterable<String> candidateLibraryNames,
     this.service, {
-    String isolateId,
+    IsolateRef isolateRef,
+    this.oneRequestAtATime = false,
   })  : _candidateLibraryNames = Set.from(candidateLibraryNames),
         _clientId = (Random.secure().nextDouble() * 10000).toInt() {
     _libraryRef = Completer<LibraryRef>();
 
     // For evals in tests, we will pass the isolateId into the constructor.
-    if (isolateId != null) {
-      _init(isolateId, false);
+    if (isolateRef != null) {
+      _init(isolateRef);
     } else {
-      selectedIsolateStreamSubscription = serviceManager.isolateManager
-          .getSelectedIsolate((IsolateRef isolate) {
-        final String id = isolate?.id;
-        _initializeComplete = null;
-        _init(id, isolate == null);
-      });
+      selectedIsolateStreamSubscription =
+          serviceManager.isolateManager.getSelectedIsolate(_init);
     }
   }
 
-  Future<void> _init(String isolateId, bool isIsolateNull) async {
-    await _initializeComplete;
+  void _init(IsolateRef isolateRef) {
```

Comment:
fix issues where these could go wrong if _init got called twice in quick succession.